### ISSUE TITLE
New note reference syntax doesn't work as documented/expected

### ISSUE THREAD

**Issue by ScriptAutomate** [dendronhq/dendron#530]
## Please select the area your Bug applys to. (Multiple selections are Possible)

- [x] UI
- [ ] Workbench
- [ ] Publishing
- [ ] Pods
- [x] Markdown
- [ ] Hierarchy
- [ ] Graph
- [ ] Misc (Choose this if your not sure)

## Describe the bug

I've been trying to figure out the new way of using reference notes, which was mentioned in a recent PR of mine (https://github.com/dendronhq/dendron-site/pull/60), as I was still using the old way. Detailed here is the newer way:
- https://dendron.so/notes/f1af56bb-db27-47ae-8406-61a98de6c78c.html#references

I didn't realize there was some cool new ways of working with note references via `![[foo]]` instead of `((ref: [[foo]]))`, but I am struggling with the actual implementation.

This format gets rendered as plain text, and isn't picked up by Dendron at build at all:

```
![[ dendron.community.showcase
    #Open AWS Catalogue,1
    :#*
]]
```

Trying to condense together at least gets rendered, but doesn't work as expected:

```
![[dendron.community.showcase#Open AWS Catalogue,1:#*]]
```

I get the on-page rendered error of: `Start anchor Open AWS Catalogue not found` (but it does exist!)

The old format still works:

```
((ref:[[dendron.community.showcase]]#Open AWS Catalogue,1:#*]]))
```

After playing around, I realized the new format works **only when there is no whitespace in a header reference, and the whitespace is replaced with a `-`**. This means that this rendered properly for the `Open AWS Catalogue` header:

```
![[dendron.community.showcase#Open-AWS-Catalogue,1:#*]]
```

## To Reproduce

I was testing this out by simply trying to convert old `((ref: [[...]]))` usage on `dendron-site`, which is why I was using the `Open AWS Catalogue` example. The page I was attempting to with updating:

- https://dendron.so/notes/ed4162aa-5474-4483-9fbc-c0b505310c59.html#other-3
  - Source: https://github.com/dendronhq/dendron-site/blob/master/vault/dendron.release.changelog.0-1-x.md#seeds

## Expected behavior

Two things.

1. Either this should work:

```
![[dendron.community.showcase#Open AWS Catalogue,1:#*]]
```

Or the documentation should be updated to push this format:

```
![[dendron.community.showcase#Open-AWS-Catalogue,1:#*]]
```

2. Some version of this should work (either with hyphens or with whitespace in header):

```
![[ dendron.community.showcase
    #Open AWS Catalogue,1
    :#*
]]
```

Or the documentation should be updated to ensure it's always a single line item.

### Code syntax testing for note refs

![image](https://user-images.githubusercontent.com/5951023/110722597-2e9ea500-81d8-11eb-8bbd-b3eac74c9daf.png)

### Rendered and unrendered note refs

![image](https://user-images.githubusercontent.com/5951023/110722655-42e2a200-81d8-11eb-8537-374e7b88921c.png)

## Desktop (please complete the following information)

- OS: Pop!_OS 20.10
- Browser: Firefox 86.0 (64-bit) || Chromium Version 89.0.4389.82 (Official Build) (64-bit)
- Dendron Version:
  ```json
  "@dendronhq/dendron-11ty": "^1.32.0",
  "@dendronhq/dendron-cli": "^0.32.0"
  ```

## Dendron Log file

I can provide log output by request, but I believe this should be easily repeatable.

**Comment by kevinslin** [comment]
thanks for the issue.

we need to update the docs. the new style note refs use github style slugging
this is because old note refs weren't able to handle special characters in the headers
tldr: whitespace is no longer supported because github-slugger doesn't support it. if you use the cmd+shift+r shortcut on a header, we will automatically sluggify the note ref

**Comment by ScriptAutomate** [comment]
Okay, I'll take on this issue and update the docs.

**Comment by ScriptAutomate** [comment]
Closing this issue. Opened https://github.com/dendronhq/dendron-site/issues/65
